geoips/plugins/modules/readers/seviri_hrit.py

```diff
@@ -183,19 +184,19 @@ def compare_dicts(d1, d2, skip=None):
 
 def get_top_level_metadata(fnames, sect):
     """Get top level metadata."""
+    md = {}
     for fname in fnames:
         df = HritFile(fname)
         if "block_2" in df.metadata.keys():
             break
-    md = {}
-    if "GEOS" in df.metadata["block_2"]["projection"]:
+    if "GEOS" in df.metadata.get("block_2", {}).get("projection", {}):
         md["sector_name"] = "Full-Disk"
-    else:
-        raise HritError(
-            "Unknown projection encountered: {}".format(
-                df.metadata["block_2"]["projection"]
-            )
-        )
+    # else:
+    # raise HritError(
+    #     "Unknown projection encountered: {}".format(
+    #         df.metadata.get("block_2", {}).get("projection", {})
+    #     )
+    # )
```

Comment:
Does this reader work with projections besides GEOS now?  Or I guess, did it always work with othre projections, we were just erroring if it was now GEOS?  I don't see any other changes to make it support other projections? @evrose54

Reply:
I'll have to look at this one. I'm forgetting why I had that problem. Might have been related to incorrect files being provided when I was testing. I'll get back to you on this tomorrow.

Reply:
Ok I figured this one out. It seems as if the new files we provided in ``seviri.WV-Upper.unprojected_image.sh`` don't have ``block_2`` an any of the files' metadata. Shown below is the contents of the last file's metadata.

So, it looks like we can set sector name to ``Full-Disk`` if we find it, or we can add a match, case statement to set ``md["sector_name"] based on the incoming blocks provided.
```
{'block_0': {'block_num': 0,
  'block_length': 16,
  'type_code': 129,
  'header_length': 90,
  'data_length': 3042600},
 'block_4': {'block_num': 4,
  'block_length': 64,
  'annotation': 'H-000-MSG2__-MSG2_IODC___-_________-EPI______-202312110800-__'},
 'block_5': {'block_num': 5,
  'block_length': 10,
  'time': b'@^\x15\x01\xc5W\xea'}}
```

Reply:
Do we need to open an issue for this, link to this conversation, then resolve this conversation?

Reply:
We can either create an issue or I can modify the code to pass in all files instead of one for certain readers. Per @AJThorpe's comment, other readers such as VIIRS SDR require multiple files. So does ahi_hsd I think. While this doesn't cause any problems because we loop over all of the files and eventually get the data we need, if we think it would be beneficial to send all files of one time step rather than a series of files at the same time step, I can make the appropriate changes. Let me know what you think here @mindyls and @jsolbrig.

Reply:
Might be more efficient to implement what I said above as well. But I'd have to test that out.

Reply:
I implemented the change I talked about two comments above. This solved metadata issues and made the readers only grab the data they needed for the incoming scan time(s). This is reflected in the most recent commits to this branch. Note that the above code still needs to be commented out, as we need to call the updated readers with each file individually the first time around to gather their date times, sort by that datetime, then provide sets of data that share unique date times. Hopefully that made sense, lol.

Reply:
Some background on why this is here. SEVIRI has two scan modes. It has a 15-minute full-disk scan and a 5-minute "extended local area" (ELA) scan. This reader is only intended to support the 15-minute full-disk imagery and will not work for the 5-minute (ELA) imagery. This test is here to ensure that we don't try to read and use ELA data.

I've been staring at this for a while and have restarted this comment a few times. I think that this would actually work if the commented section were to be uncommented. I could be wrong, though, and don't have time to test right now. I'll try to look at this tonight or tomorrow.

Essentially, though, `get_top_level_metadata` is called with a list of files, one of which MUST be a prologue file. Prologue files contain `block_2` but none of the other files do. So, it loops through the provided file to find the one that contains `block_2`, then collects the projection metadata from that file.

If this test isn't possible here, we need to implement it somewhere.

Reply:
I found a workaround! It's a little bit black magic, but it does the trick. Essentially, I'll add start datetime and end datetime to the error, parse that out, recreate the date times, and use them within ``ReadersInterface.read_data_to_xarray_dict`` where needed. The next call of the reader, which provides files by unique scan times, will raise an HritError which will not be caught if in fact all of the files are using the wrong projection.